### PR TITLE
Multi-batch unit tests for Lie groups

### PR DESCRIPTION
**Pull request by luisenp**
Missing unit test for gradient ops

### CODE REVIEW

File: tests/labs/lie/functional/common.py [hunk 1]
```diff
@@ -56,57 +57,81 @@ def get_test_cfg(op_name, dtype, dim, data_shape, module=None):
...
+#   - multi-batch output consistent with single-batch output
 def run_test_op(op_name, batch_size, dtype, rng, dim, data_shape, module):
+    is_multi_batch = not isinstance(batch_size, int)
+    bs = len(batch_size) if is_multi_batch else 1
     all_input_types, atol = get_test_cfg(op_name, dtype, dim, data_shape, module=module)
     for input_types in all_input_types:
         inputs = sample_inputs(input_types, batch_size, dtype, rng)
         funcs = (
-            tuple(left_project_func(module, x) for x in inputs)
+            tuple(left_project_func(module, x, bs) for x in inputs)
```
Comment (fantaosha) [inline tests/labs/lie/functional/common.py:103]: Wondering if we need to check `bs=None` or `dim_out=None`, which assumes no broadcasting and is most common user-case.

File: tests/labs/lie/functional/common.py [hunk 2]
```diff
@@ -122,17 +147,52 @@ def check_lie_group_function(module, op_name: str, atol: float, inputs, funcs=No
...
+        # Convert to single-batch-dim sparse gradient format
+        batch_size = matrix.shape[:batch_dim]
+        if batch_dim > 0:
+            d = reduce(lambda x, y: x * y, batch_size)
+            matrix = matrix.reshape(d, -1, d, *group.shape[-2:])
+            sels = range(matrix.shape[0])
+            matrix = matrix[sels, ..., sels, :, :]
+            g = group.reshape(d, *group.shape[-2:])
+        # Compute projected gradient matrix
+        ret = module._left_project_autograd_fn(g, matrix)
```
Comment (fantaosha) [inline tests/labs/lie/functional/common.py:179]: Do we assume the same dimension of inputs and outputs?

**Review by fantaosha — APPROVED** [review]
LGTM. Only minor comments.

File: tests/labs/lie/functional/common.py (hunk 1 above)
Comment (luisenp) [reply tests/labs/lie/functional/common.py:103]: I thought the case `tuple()` covered this for the no batch size (see line 10). Or do you mean something else? 

The tests for different `dim_out` are in #527.

File: tests/labs/lie/functional/common.py (hunk 2 above)
Comment (luisenp) [reply tests/labs/lie/functional/common.py:179]: Yes, I think so. There should be no broadcasting in the test where this function is used.

**Pull request merged**